dot/core/messages.go

```diff
@@ -6,15 +6,57 @@ package core
 import (
 	"errors"
 
-	"github.com/libp2p/go-libp2p-core/peer"
-
 	"github.com/ChainSafe/gossamer/dot/network"
 	"github.com/ChainSafe/gossamer/dot/peerset"
 	"github.com/ChainSafe/gossamer/dot/types"
 	"github.com/ChainSafe/gossamer/lib/runtime"
 	"github.com/ChainSafe/gossamer/lib/transaction"
+
+	"github.com/libp2p/go-libp2p-core/peer"
 )
 
+func (s *Service) validateTransaction(peerID peer.ID, head *types.Header, rt runtime.Instance,
+	tx types.Extrinsic, toPropagate *[]types.Extrinsic) (bool, error) {
+	s.storageState.Lock()
+	defer s.storageState.Unlock()
+
+	ts, err := s.storageState.TrieState(&head.StateRoot)
+	if err != nil {
+		return true, err
+	}
+
+	rt.SetContextStorage(ts)
+
+	// validate each transaction
+	externalExt := types.Extrinsic(append([]byte{byte(types.TxnExternal)}, tx...))
+	val, err := rt.ValidateTransaction(externalExt)
+	if err != nil {
+		if errors.Is(err, runtime.ErrInvalidTransaction) {
+			s.net.ReportPeer(peerset.ReputationChange{
+				Value:  peerset.BadTransactionValue,
+				Reason: peerset.BadTransactionReason,
+			}, peerID)
+		}
+
+		logger.Debugf("failed to validate transaction: %s", err)
+		return true, nil
```

Comment:
What's the boolean returned here?

Reply:
As in it is as intuitive as the other way. It's just Go standards have disformed my brain so when something returns a boolean that is `not Something` it triggers me because I really don't want to have `!notSomething` 😄